### PR TITLE
Organising your data (aka. groups)

### LINKED ISSUE #342: 👌 IMPROVE: Module: Data - Organising your data
The module on "Organising your data" discusses:

* How to create, show and manipulate groups.
* How to add/remove nodes to/from groups.
* How to organise groups in hierarchies.

The content is already in good shape, so the main point of this issue is to go over the material and see if it can still be improved somehow. Are the explanations clear? Would you add/remove something? Does the material adhere to the [principles of a tutorial](https://documentation.divio.com/tutorials/)? That sort of thing.

Although testing the tutorial content is of course fine, note that we'll do a full round of testing in the week of the 21st.

### PR DESCRIPTION
**Pull request by CasperWA**
Closes #342 

I've split this into 3 commits (for review purposes only):

1. Improving the actual content - not much to do here, just very minor clarifications.
2. Remove empty lines in code blocks.
3. Try to update the text according to #360.

P.S. I can't find information about who is in who's sections from the Google Drive, so I just put the @mbercx and @chrisjsewell as reviewers here...

### CODE REVIEW

File: docs/sections/data/groups.md [hunk 1]
```diff
@@ -71,54 +67,44 @@ PK    Type         Created
...
 ...
-
 ```
 
-Conversely, if you want to see all the groups a node belongs to, you can run:
+Conversely, if you want to see all the groups a node belongs to, you can use its PK and run:
 
 ```{code-block} console
-
 $ verdi group list -a -A --node 380
```
Comment (mbercx) [inline docs/sections/data/groups.md]: I'll use this opportunity to ask you what you think on https://github.com/aiidateam/aiida-tutorials/issues/378 😄

File: docs/sections/data/groups.md [hunk 2]
```diff
@@ -71,54 +67,44 @@ PK    Type         Created
...
 PK    Label            Type string    User
 ----  ---------------  -------------  ---------------
    1  tutorial_pbesol  core           aiida@localhost
    7  20200705-071658  core.import    aiida@localhost
-
 ```
 
 ### Creating and manipulating groups
 
 Lets make a new group:
```
Comment (mbercx) [inline docs/sections/data/groups.md]:
```suggestion
Let's make a new group:
```

File: docs/sections/data/groups.md [hunk 3]
```diff
@@ -71,54 +67,44 @@ PK    Type         Created
...
 
 ```{code-block} console
-
 $ verdi group create a_group
 Success: Group created with PK = 8 and name 'a_group'
-
 ```
 
-If we want to change the name of the group at any time:
+If you want to change the name of the group at any time try:
```
Comment (mbercx) [inline docs/sections/data/groups.md]:
```suggestion
You can change the name of a group using the `verdi group relabel` command:
```

File: docs/sections/data/groups.md [hunk 4]
```diff
@@ -71,54 +67,44 @@ PK    Type         Created
...
 
 ```{code-block} console
-
 $ verdi group relabel a_group my_group
 Success: Label changed to my_group
-
 ```
 
-Now we can add one or more nodes to it:
+Now you can add one or more nodes to it by listing any number of node PKs:
```
Comment (mbercx) [inline docs/sections/data/groups.md]: Lot's of optional-sounding "can" phrases are used here. No big deal, but maybe we can also use the _imperative mood_ to give instructions instead? (see the examples in #360 ). The "can" suggestions in the React tutorial are _really_ suggestions, but here it feels more like we're giving instructions. The sentence would become:

```suggestion
Add one or more nodes to your new group using node PKs from the `tutorial_pbesol` group we inspected earlier:
```

File: docs/sections/data/groups.md [hunk 5]
```diff
@@ -71,54 +67,44 @@ PK    Type         Created
...
 
 ```{code-block} console
-
 $ verdi group add-nodes -G my_group 380 1273
 Do you really want to add 2 nodes to Group<my_group>? [y/N]: y
-
 ```
 
-We can also copy the nodes from an existing group to another group:
+You can also copy the nodes from an existing group to another group:
```
Comment (mbercx) [inline docs/sections/data/groups.md]:
```suggestion
You can also copy *all* nodes from an existing group to another group using:
```

File: docs/sections/data/groups.md [hunk 6]
```diff
@@ -135,31 +121,26 @@ PK    Type         Created
 74  CalcJobNode  2078D:17h:51m ago
 76  CalcJobNode  2078D:17h:57m ago
 ...
-
 ```
 
-To remove nodes from the group run:
+To remove nodes from the group is similar to adding them, try:
```
Comment (mbercx) [inline docs/sections/data/groups.md]:
```suggestion
Removing nodes from the group is similar to adding them, try:
```

File: docs/sections/data/groups.md [hunk 7]
```diff
@@ -168,65 +149,57 @@ The same happens if you remove nodes from the group -- they will remain in the d
 ## Organising groups in hierarchies
 
 Earlier we mentioned that groups are like files in folders on your filesystem.
-As with folders and sub-folders then, as the amount of groups we have grows, we may also wish to structure our groups in a hierarchy.
-Groups in AiiDA are inherently "flat", in that groups may only contain nodes and not other groups.
-However it is possible to construct *virtual* group hierarchies based on delimited group labels, using the `grouppath` utility.
+Then as with folders and sub-folders, as the amount of groups you have grows, you may also wish to structure your groups in a hierarchy.
```
Comment (mbercx) [inline docs/sections/data/groups.md]:
```suggestion
Similar to folders and sub-folders, you may wish to structure your groups in a hierarchy once the number of groups in your database grows larger.
```

File: docs/sections/data/groups.md [hunk 8]
```diff
@@ -168,65 +149,57 @@ The same happens if you remove nodes from the group -- they will remain in the d
...
+Groups in AiiDA are inherently "flat", meaning groups may only contain nodes and not other groups.
+However, it is possible to construct *virtual* group hierarchies based on delimited group labels, using the `grouppath` utility.
 
-Like folder paths grouppath requires delimitation by `/` (forward slash) characters.
+Like folder paths on Unix systems `grouppath` requires delimitation by forward slash (`/`) characters.
```
Comment (mbercx) [inline docs/sections/data/groups.md]:
```suggestion
Like folder paths on Unix systems, `grouppath` requires delimitation by forward slash (`/`) characters.
```

File: docs/sections/data/groups.md [hunk 9]
```diff
@@ -168,65 +149,57 @@ The same happens if you remove nodes from the group -- they will remain in the d
...
 Lets copy and rename the three tutorial groups:
```
Comment (mbercx) [inline docs/sections/data/groups.md]:
```suggestion
Let's copy and rename the three tutorial groups:
```

File: docs/sections/data/groups.md [hunk 10]
```diff
@@ -168,65 +149,57 @@ The same happens if you remove nodes from the group -- they will remain in the d
...
 tutorial_lda                0
 tutorial_pbe                0
 tutorial_pbesol             0
-
 ```
 
 :::{note}
 
-In the terminal, paths that contain nodes are listed in bold
+In the terminal, paths that contain nodes are listed in bold.
```
Comment (mbercx) [inline docs/sections/data/groups.md:179]: Is this true? I don't seem to notice this. 🤔

File: docs/sections/data/groups.md [hunk 11]
```diff
@@ -168,65 +149,57 @@ The same happens if you remove nodes from the group -- they will remain in the d
...
 
 :::
 
-You can see that the actual groups that we create do not show, only the initial part of the "path", and how many sub-groups that path contains.
-We can then step into a path:
+You can see that the actual groups that are created do not appear, only the initial part of the "path", and how many sub-groups that path contains.
```
Comment (mbercx) [inline docs/sections/data/groups.md]:
```suggestion
You can see that the actual groups we just created do not appear, only the initial part of the path (`tutorial`), and how many subgroups that path contains.
```

**Review by mbercx — CHANGES_REQUESTED** [review]
Thanks @CasperWA! 🙏  I've given the section a pass and left some suggestions/comments. Since we need to merge this tomorrow (or rather, _today_ 😅 ), feel free to bypass comments that you think will take too much time to fix.

File: docs/sections/data/groups.md [hunk 12]
```diff
@@ -3,15 +3,14 @@
 # Organising your data
 
 In this section of the tutorial we will focus on how to organise and explore the data in an AiiDA database.
-As in (**TODO FIX LINK**), we will be using the previously created database entries for this tutorial.
-To follow the tutorial then, you can use the profile in to which you have previously imported this data, or you may wish to create a fresh profile and import the archive into that:
+You will be using the previously created database entries for this tutorial.
```
[the remark this answers is not in the document]
Comment (ramirezfranciscof) [reply docs/sections/data/groups.md]: Mmm may I suggest rephrasing further? On many reads of this I interpreted this to mean "you will be using the entries you created in previous sections of this tutorial". Maybe changing the order at least? But I would even be more explicit, something like:

```suggestion
For this tutorial you will be loading and using the information of an external pre-existing database.
```

[the remark this answers is not in the document]
Comment (mbercx) [reply docs/sections/data/groups.md]: Sure, all good! Let's not be too nit-picky at this stage though, we need to get this merged asap. ;D

File: docs/sections/data/groups.md [hunk 13]
```diff
@@ -3,15 +3,14 @@
...
 
 ```{code-block} console
-
-$ verdi quicksetup --profile data
-$ verdi profile setdefault data
-$ verdi import https://object.cscs.ch/v1/AUTH_b1d80408b3d340db9f03d373bbde5c1e/marvel-vms/tutorials/aiida_tutorial_2020_07_perovskites_v0.9.aiida
-
+verdi quicksetup --profile data
+verdi profile setdefault data
+verdi import https://object.cscs.ch/v1/AUTH_b1d80408b3d340db9f03d373bbde5c1e/marvel-vms/tutorials/aiida_tutorial_2020_07_perovskites_v0.9.aiida
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/sections/data/groups.md]: Am I presenting this or just the main "overseer" when they go through it?

[the remark this answers is not in the document]
Comment (mbercx) [reply docs/sections/data/groups.md]: Again, this depends on your style. But I would at least give an overview at the start? You'll also have to explain that there are two modules in the "Managing Data" hands-on session.

File: docs/sections/data/groups.md [hunk 14]
```diff
@@ -37,28 +35,26 @@ PK    Label            Type string    User
...
 
 The default table shows us four pieces of information:
 
-* **PK**: The Primary Key of the group
-* **Label**: The label by which the group can be identified
-* **Type string**: This tells us what "sub-class" of group this is.
+* **PK**: The Primary Key of the group.
+* **Label**: The label by which the group can be identified.
+* **Type string**: This tells us what type of group this is.
   Type strings can be used to class certain types of data, for example here we have general groups (`core`), groups containing pseudopotentials (`core.upf`), and an auto-generated group containing the nodes we imported from the archive (`core.import`).
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/sections/data/groups.md:45]: I'll leave this up to you to do at this point... sorry.

[the remark this answers is not in the document]
Comment (mbercx) [reply docs/sections/data/groups.md:45]: No worries! 👍  I'll give the section another pass when testing anyways

File: docs/sections/data/groups.md (hunk 10 above)
Comment (CasperWA) [reply docs/sections/data/groups.md:179]: I can't actually check it now, but the code has `bold=True`? https://github.com/aiidateam/aiida-core/blob/develop/aiida/cmdline/commands/cmd_group.py#L448

Comment (mbercx) [reply docs/sections/data/groups.md:179]: Well, i'll double check it during my pass, thanks!

**Comment by CasperWA** [comment]
Thanks for the review @mbercx - I'll let you handle it from here as I'm out of time :/

**Comment by mbercx** [comment]
> Thanks for the review @mbercx - I'll let you handle it from here as I'm out of time :/

image

**Review by mbercx — APPROVED** [review]
Let's get this show on the road! 🚀  Thanks for revising this section, @CasperWA!

**Pull request merged**